src/autopas/containers/hierarchicalGrids/HierarchicalGrids.h

```diff
@@ -0,0 +1,644 @@
+
+        // Iterate pairwise over the cross-level
+        crossLevel.iteratePairwise(traversal);
+
+        // Move particles back to their level
+        AUTOPAS_OPENMP(parallel)
+        for (auto iterCross = crossLevel.begin(); iterCross.isValid(); ++iterCross) {
+
+          const double particleRad = iterCross->getRad();
+
+          //if targetLevel = largerLevel, particle has negative radius. Hence, reverse the ugly trick
+          if (particleRad < 0.) {
+            
+            iterCross->setRad(std::abs(particleRad));
+          }
+
+          unsigned int targetLevel = getHierarchyLevelOfParticle(*iterCross);
+
+          addParticleToGridLevel(*iterCross, targetLevel);
+        }
+        
+        crossLevel.deleteAllParticles();
+
+      }
+    }
+
+    demLib::DEMFunctor<Particle>::endCrossLevelTraversal();
+  }
+
+  /**
+   * @todo
+   * 
+   * Iterates over all particle pairs in the container.
+   * @param traversal The traversal to use for the iteration.
+   */
+  void iteratePairwise(TraversalInterface *traversal) override {
+
+    // Distribute particles over the hierarchical grid levels, if more than one HG level is present
+    if (_numberOfLevels > 1) {
+      
+      distributeParticlesToHGLevels();
+
+    }
+    // Else skip the distribution and just iteratePairwise over the carrier level and return
+    else {
+      
+      _carrierLevel.iteratePairwise(traversal);
+
+      return;
+
+    }
+
+    // Iterate over hierarchy levels 
+    iterateAllHGLevels(traversal);
+
+    // Iterate over cross levels 
+    iterateAllCrossLevels(traversal);
+
+    
+    // Distribute particles back to the top level hierarchical grid
+    distributeParticlesToCarrierLevel();
+
+  }
+
+  /**
+   * @copydoc autopas::ParticleContainerInterface::getBoxMax()
+   */
+  [[nodiscard]] const std::array<double, 3> &getBoxMax() const final { return _boxMax; }
+
+  /**
+   * @copydoc autopas::ParticleContainerInterface::getBoxMin()
+   */
+  [[nodiscard]] const std::array<double, 3> &getBoxMin() const final { return _boxMin; }
+
+  /**
+   * @copydoc autopas::ParticleContainerInterface::getCutoff()
+   */
+  [[nodiscard]] double getCutoff() const final { return _cutoff; }
+
+  /**
+   * @copydoc autopas::ParticleContainerInterface::setCutoff()
+   */
+  void setCutoff(double cutoff) final { _cutoff = cutoff; }
+
+  /**
+   * @copydoc autopas::ParticleContainerInterface::getVerletSkin()
+   */
+  [[nodiscard]] double getVerletSkin() const override { return _skinPerTimestep * _rebuildFrequency; }
+
+  /**
+   * @copydoc autopas::ParticleContainerInterface::getInteractionLength()
+   */
+  [[nodiscard]] double getInteractionLength() const override { return _cutoff + _skinPerTimestep * _rebuildFrequency; }
+
+  /**
+   * @TODO !!! Funktioniert so bestimmt nicht ... 
+   * 
+   * @copydoc autopas::ParticleContainerInterface::updateContainer()
+   */
+  [[nodiscard]] std::vector<ParticleType> updateContainer(bool keepNeighborListsValid) {
```

Comment:
```suggestion
  [[nodiscard]] std::vector<ParticleType> updateContainer(bool keepNeighborListsValid) override {
```

check if this is missing somewhere else too

Reply:
Done